src/ui/public/visualize/loader/__tests__/visualize_loader.js

```diff
@@ -297,26 +298,30 @@ describe('visualize loader', () => {
             added: 'value',
           }
         });
-        // Sync we are relying on $evalAsync we need to trigger a digest loop during tests
-        $rootScope.$digest();
         expect(container.find('[data-test-subj="visualizationLoader"]')[0].hasAttribute('data-foo')).to.be(false);
         expect(container.find('[data-test-subj="visualizationLoader"]').attr('data-added')).to.be('value');
       });
 
-      it('should allow updating the time range of the visualization', () => {
+      it('should allow updating the time range of the visualization', async () => {
+        const spy = sinon.spy(VisualizeDataLoader.prototype, 'fetch');
+
         const handler = loader.embedVisualizationWithSavedObject(newContainer()[0], createSavedObject(), {
           timeRange: { from: 'now-7d', to: 'now' }
         });
+
+        // Wait for the initial fetch and render to happen
+        await timeout(150);
```

Comment:
I'm sure there is a reason for using these timeouts here, but I can't not comment on it. It looks like a flaky test in the making. :wink: Is there no way to synchronously force the "fetch and render to happen"? Maybe the time mocking functionality of sinon can be of use?

Reply:
Unfortunately we currently cannot (I think I left a comment in the timeout below), since we use a too old `lodash` version. Basically we need to wait for the `debounce` to happen. Lodash prior 4, does not properly work together with jest or sinon fake timers, so that's the most reasonable way to trigger it imho. Also since we don't wait for something "arbitrary long" I don't think that will ever introduce flakyness, but we just need the debounce to be triggered. As soon as we update to lodash 4, we can use `sinon.useFakeTimers()` instead to make this sync.

Reply:
How do you know that the debounce has always happened after 150ms? That looks suspiciously arbitrary. :innocent: waiting for the "next tick" like in the prior test cases is not sufficient?

Reply:
Unfortunately that's not enough, since in that case we actually need to wait for the debounce to be done, while the above doesn't care about that. After an update to lodash 4 that might be better, since we don't need to give it an arbitrary number here anymore, but can "flush" all pending debounce operations.